Null pointer exception on creating LDAP user entry

If you click on "Write LDAP configuration" after you added a new user but did forget to select the correct LDAP user group, you get a null pointer exception:

```
[ERROR] 2021-06-14 12:59:27,665 [http-nio-127.0.0.1-8080-exec-1] org.kitodo.production.forms.UserForm - Could not generate ldap entry
java.lang.NullPointerException: null
        at org.kitodo.production.services.data.LdapServerService.createNewUser(LdapServerService.java:160) ~[classes/:3.3]
        at org.kitodo.production.forms.UserForm.writeUserAtLdapServer(UserForm.java:489) [classes/:3.3]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_292]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_292]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_292]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_292]
        at org.apache.el.parser.AstValue.invoke(AstValue.java:247) [tomcat8-jasper-el-8.5.54.jar:8.5.54]
        at org.apache.el.MethodExpressionImpl.invoke(MethodExpressionImpl.java:267) [tomcat8-jasper-el-8.5.54.jar:8.5.54]
        at org.jboss.weld.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
        at org.jboss.weld.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
        at org.jboss.weld.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
        at org.jboss.weld.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50) [weld-servlet-2.4.8.Final.jar:2.4.8.Final]
        at org.apache.myfaces.view.facelets.el.ContextAwareTagMethodExpression.invoke(ContextAwareTagMethodExpression.java:96) [myfaces-impl-2.3.4.jar:2.3.4]
        at org.apache.myfaces.application.ActionListenerImpl.processAction(ActionListenerImpl.java:74) [myfaces-impl-2.3.4.jar:2.3.4]
        at org.springframework.faces.webflow.FlowActionListener.processAction(FlowActionListener.java:71) [spring-faces-2.5.1.RELEASE.jar:2.5.1.RELEASE]
        at org.springframework.faces.model.SelectionTrackingActionListener.processAction(SelectionTrackingActionListener.java:64) [spring-faces-2.5.1.RELEASE.jar:2.5.1.RELEASE]
        at javax.faces.component.UICommand.broadcast(UICommand.java:120) [myfaces-api-2.3.4.jar:2.3.4]
        at javax.faces.component.UIViewRoot._broadcastAll(UIViewRoot.java:1255) [myfaces-api-2.3.4.jar:2.3.4]
        at javax.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:420) [myfaces-api-2.3.4.jar:2.3.4]
        at javax.faces.component.UIViewRoot._process(UIViewRoot.java:1741) [myfaces-api-2.3.4.jar:2.3.4]
        at javax.faces.component.UIViewRoot.processDecodes(UIViewRoot.java:942) [myfaces-api-2.3.4.jar:2.3.4]
        at org.apache.myfaces.lifecycle.ApplyRequestValuesExecutor.execute(ApplyRequestValuesExecutor.java:42) [myfaces-impl-2.3.4.jar:2.3.4]
        at org.apache.myfaces.lifecycle.LifecycleImpl.executePhase(LifecycleImpl.java:195) [myfaces-impl-2.3.4.jar:2.3.4]
        at org.apache.myfaces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:142) [myfaces-impl-2.3.4.jar:2.3.4]
        at javax.faces.webapp.FacesServlet.service(FacesServlet.java:204) [myfaces-api-2.3.4.jar:2.3.4]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) [tomcat8-catalina-8.5.54.jar:8.5.54]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [tomcat8-catalina-8.5.54.jar:8.5.54]
...
```

There is no check in user frontend which is checking is a ldap group selected for this user nor is in code a check if for a user a ldap group assigned.

--- a/Kitodo/src/main/java/org/kitodo/production/services/data/LdapServerService.java
+++ b/Kitodo/src/main/java/org/kitodo/production/services/data/LdapServerService.java
@@ -157,6 +157,11 @@ public class LdapServerService extends SearchDatabaseService<LdapServer, LdapSer
     public void createNewUser(User user, String password)
             throws NamingException, NoSuchAlgorithmException, IOException {
 
+        if (Objects.isNull(user.getLdapGroup())) {
+            Helper.setMessage(Helper.getTranslation("noLdapGroupAssignedToUser"));
+            return;
+        }
+
         if (!user.getLdapGroup().getLdapServer().isReadOnly()) {
             Hashtable<String, String> ldapEnvironment = initializeWithLdapConnectionSettings(
                 user.getLdapGroup().getLdapServer());

--- a/Kitodo/src/main/resources/messages/messages_de.properties
+++ b/Kitodo/src/main/resources/messages/messages_de.properties
@@ -689,6 +689,7 @@ newWorkflow=Neuer Workflow
 next=Weiter
 no=Nein
 noExistingWorkflow=Es ist noch kein Workflow im System
+noLdapGroupAssignedToUser=Dem Benutzer ist keine LDAP Gruppe zugewiesen
 noMedia=Keine Medien
 none=Keine
 normDataRecord=URI

--- a/Kitodo/src/main/resources/messages/messages_en.properties
+++ b/Kitodo/src/main/resources/messages/messages_en.properties
@@ -701,6 +701,7 @@ newUser=New user
 next=Next
 no=no
 noExistingWorkflow=No workflow found
+noLdapGroupAssignedToUser=No LDAP group is assigned to user
 noMedia=No media
 none=None
 normDataRecord=URI\

--- a/Kitodo/src/main/webapp/WEB-INF/templates/includes/userEdit/details.xhtml
+++ b/Kitodo/src/main/webapp/WEB-INF/templates/includes/userEdit/details.xhtml
@@ -62,14 +62,13 @@
                                          value="#{UserForm.userObject.active}" onchange="toggleSave()"/>
             </div>
             <div>
-                <p:commandButton id="writUserToLdapButton"
+                <p:commandButton id="writeUserToLdapButton"
                                  action="#{UserForm.writeUserAtLdapServer}"
                                  value="#{msgs.ldapKonfigurationSchreiben}"
                                  icon="fa fa-user-plus fa-lg"
                                  iconPos="right"
                                  immediate="true"
-                                 disabled="#{isViewMode || isConfigMode}"
-                                 rendered="#{AuthenticationController.ldapAuthentication}"
+                                 disabled="#{isViewMode || isConfigMode || UserForm.userObject.ldapGroup eq null}"
                                  class="button-filled-green">
                 </p:commandButton>
                 <p:commandButton onclick="PF('changePasswordDialog').show()"

